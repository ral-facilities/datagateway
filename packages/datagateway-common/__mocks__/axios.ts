import axios from 'axios';

const requests = {
  get: vi.fn(() => Promise.resolve({ data: {} })),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  delete: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  AxiosError: axios.AxiosError,
  isAxiosError: axios.isAxiosError,
};

export default requests;
