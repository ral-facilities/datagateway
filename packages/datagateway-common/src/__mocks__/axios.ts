// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn(() => Promise.resolve({ data: {} })),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  AxiosError: axios.AxiosError,
};

export default requests;
