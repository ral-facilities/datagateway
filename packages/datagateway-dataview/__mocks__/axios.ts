import axios from 'axios';

const requests = {
  get: vi.fn((path) => {
    if (path === '/datagateway-dataview-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'Generic',
          idsUrl: 'ids',
          apiUrl: 'api',
          downloadApiUrl: 'download',
          'ui-strings': '/res/default.json',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  delete: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  isAxiosError: axios.isAxiosError,
};

export default requests;
