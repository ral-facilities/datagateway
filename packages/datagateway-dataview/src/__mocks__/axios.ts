// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn((path) => {
    if (path === '/datagateway-dataview-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'Generic',
          idsUrl: 'ids',
          apiUrl: 'api',
          downloadApiUrl: 'download',
          'ui-strings': '/res/default.json',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  isAxiosError: axios.isAxiosError,
};

export default requests;
