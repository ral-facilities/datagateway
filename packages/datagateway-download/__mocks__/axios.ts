import axios from 'axios';

const requests = {
  get: vi.fn((path) => {
    if (path.includes('/topcat/user/cart/')) {
      return Promise.resolve({ data: { cartItems: [] } });
    } else if (path === '/datagateway-download-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'LILS',
          apiUrl: 'https://example.com/api',
          downloadApiUrl: 'https://example.com/downloadApi',
          idsUrl: 'https://example.com/ids',
          fileCountMax: 5000,
          totalSizeMax: 1000000000000,
          accessMethods: {
            https: {
              idsUrl: 'https://example.com/ids',
              displayName: 'HTTPS',
              description: 'Example description for HTTPS access method.',
            },
            globus: {
              idsUrl: 'https://example.com/ids',
              displayName: 'Globus',
              description: 'Example description for Globus access method.',
            },
          },
          routes: [
            {
              section: 'Test',
              link: '/download',
              displayName: 'Download',
              order: 0,
            },
          ],
        },
      });
    } else if (path.includes('/topcat/user/downloads')) {
      return Promise.resolve({ data: [] });
    } else {
      return Promise.resolve({ data: {} });
    }
  }),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  put: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};

export default requests;
