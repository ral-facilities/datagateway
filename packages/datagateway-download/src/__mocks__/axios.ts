// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn((path) => {
    if (path.includes('/topcat/user/cart/')) {
      return Promise.resolve({ data: { cartItems: [] } });
    } else if (path === '/datagateway-download-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'LILS',
          apiUrl: 'https://example.com/api',
          downloadApiUrl: 'https://example.com/downloadApi',
          idsUrl: 'https://example.com/ids',
          fileCountMax: 5000,
          totalSizeMax: 1000000000000,
          accessMethods: {
            https: {
              idsUrl: 'https://example.com/ids',
              displayName: 'HTTPS',
              description: 'Example description for HTTPS access method.',
            },
            globus: {
              idsUrl: 'https://example.com/ids',
              displayName: 'Globus',
              description: 'Example description for Globus access method.',
            },
          },
          routes: [
            {
              section: 'Test',
              link: '/download',
              displayName: 'Download',
              order: 0,
            },
          ],
        },
      });
    } else if (path.includes('/topcat/user/downloads')) {
      return Promise.resolve({ data: [] });
    } else {
      return Promise.resolve({ data: {} });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  put: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};

export default requests;
