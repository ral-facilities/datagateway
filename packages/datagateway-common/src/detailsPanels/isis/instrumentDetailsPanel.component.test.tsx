import type { RenderResult } from '@testing-library/react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { UserEvent } from '@testing-library/user-event/dist/types/setup';
import axios from 'axios';
import * as React from 'react';
import { Provider } from 'react-redux';
import { combineReducers, createStore } from 'redux';
import { StateType } from '../../../lib';
import dGCommonReducer from '../../state/reducers/dgcommon.reducer';
import { QueryClient, QueryClientProvider } from 'react-query';
import type { Instrument } from '../../app.types';
import InstrumentDetailsPanel from './instrumentDetailsPanel.component';

function renderComponent({
  rowData,
  detailsPanelResize,
}: {
  rowData: Instrument;
  detailsPanelResize?: () => void;
}): RenderResult {
  return render(
    <Provider
      store={createStore(
        combineReducers<Partial<StateType>>({ dgcommon: dGCommonReducer })
      )}
    >
      <QueryClientProvider client={new QueryClient()}>
        <InstrumentDetailsPanel
          rowData={rowData}
          detailsPanelResize={detailsPanelResize}
        />
      </QueryClientProvider>
    </Provider>
  );
}

describe('Instrument details panel component', () => {
  let rowData: Instrument;
  let user: UserEvent;

  beforeEach(() => {
    user = userEvent.setup();
    rowData = {
      id: 1,
      name: 'Test',
      fullName: 'Test Instrument',
      description: 'An instrument purely for testing',
      type: 'testing',
      url: 'www.example.com',
    };

    axios.get = jest.fn().mockResolvedValue({
      data: [rowData],
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { asFragment } = renderComponent({ rowData });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show default tab on first render', async () => {
    renderComponent({ rowData });
    expect(
      await screen.findByRole('tab', { name: 'instruments.details.label' })
    ).toHaveAttribute('aria-selected', 'true');
  });

  it('should render users tab when present in the data', () => {
    rowData.instrumentScientists = [
      {
        id: 4,
        user: {
          id: 5,
          name: 'Louise',
          fullName: 'Louise Davies',
        },
      },
      {
        id: 9,
        user: {
          id: 10,
          name: 'Louise',
        },
      },
    ];

    const { asFragment } = renderComponent({ rowData });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should let user switch between tabs', async () => {
    rowData.instrumentScientists = [
      {
        id: 4,
        user: {
          id: 5,
          name: 'Rick',
          fullName: 'Rick Ashley',
        },
      },
    ];

    renderComponent({ rowData });

    expect(
      await screen.findByRole('tabpanel', { name: 'instruments.details.label' })
    ).toBeVisible();

    // switch tab
    await user.click(
      await screen.findByRole('tab', {
        name: 'instruments.details.instrument_scientists.label',
      })
    );

    expect(
      await screen.findByRole('tabpanel', {
        name: 'instruments.details.instrument_scientists.label',
      })
    ).toBeVisible();
  });

  it('should call detailsPanelResize on load and when tabs are switched between', async () => {
    rowData.instrumentScientists = [
      {
        id: 4,
        user: {
          id: 5,
          name: 'Louise',
          fullName: 'Louise Davies',
        },
      },
    ];

    const mockDetailsPanelResize = jest.fn();

    renderComponent({
      rowData,
      detailsPanelResize: mockDetailsPanelResize,
    });

    await waitFor(() => {
      expect(mockDetailsPanelResize).toHaveBeenCalledTimes(1);
    });

    await user.click(
      await screen.findByRole('tab', {
        name: 'instruments.details.instrument_scientists.label',
      })
    );

    await waitFor(() => {
      expect(mockDetailsPanelResize).toHaveBeenCalledTimes(2);
    });
  });

  it('should gracefully handle InstrumentScientists without Users', () => {
    rowData.instrumentScientists = [
      {
        id: 4,
      },
    ];

    const { asFragment } = renderComponent({ rowData });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render users tab and text "No Scientists" when no data is present', async () => {
    rowData.instrumentScientists = [];
    renderComponent({ rowData });
    expect(
      await screen.findByText(
        'instruments.details.instrument_scientists.no_name'
      )
    ).toBeInTheDocument();
  });

  it('should show "No <field> provided" incase of a null field', async () => {
    const { description, type, url, ...amendedRowData } = rowData;
    axios.get = jest.fn().mockResolvedValue({
      data: [amendedRowData],
    });

    renderComponent({ rowData: amendedRowData });

    expect(
      await screen.findByText('instruments.details.description not provided')
    ).toBeInTheDocument();
    expect(
      screen.getByText('instruments.details.type not provided')
    ).toBeInTheDocument();
    expect(
      screen.getByText('instruments.details.url not provided')
    ).toBeInTheDocument();
  });

  it('should display shortened instrument name if full name not provided', async () => {
    rowData = {
      id: 1,
      name: 'My test instrument',
      description: 'An instrument purely for testing',
      type: 'testing',
      url: 'www.example.com',
    };

    renderComponent({ rowData });

    expect(await screen.findByText('My test instrument')).toBeInTheDocument();
  });
});
