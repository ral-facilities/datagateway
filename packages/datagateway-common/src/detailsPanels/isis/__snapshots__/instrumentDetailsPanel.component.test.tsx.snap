// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instrument details panel component should gracefully handle InstrumentScientists without Users 1`] = `
<DocumentFragment>
  <div
    id="details-panel"
    style="min-width: 0;"
  >
    <div
      class="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
        style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
      />
      <div
        class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
        style="margin-bottom: 0px;"
      >
        <div
          aria-label="instruments.details.tabs_label"
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-controls="instrument-details-panel"
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
            id="instrument-details-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            instruments.details.label
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="instrument-users-panel"
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
            id="instrument-users-tab"
            role="tab"
            tabindex="-1"
            type="button"
          >
            instruments.details.instrument_scientists.label
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-11yukd5-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      aria-labelledby="instrument-details-tab"
      hidden=""
      id="instrument-details-panel"
      role="tabpanel"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-jddpo1-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            <b>
              Test Instrument
            </b>
          </h6>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-k7lt0k-MuiDivider-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            instruments.details.description
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              An instrument purely for testing
            </b>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            instruments.details.type
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              testing
            </b>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            instruments.details.url
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="www.example.com"
              >
                www.example.com
              </a>
            </b>
          </p>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="instrument-users-tab"
      hidden=""
      id="instrument-users-panel"
      role="tabpanel"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-jddpo1-MuiGrid-root"
      >
        <span
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          instruments.details.instrument_scientists.name
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Instrument details panel component should render correctly 1`] = `
<DocumentFragment>
  <div
    id="details-panel"
    style="min-width: 0;"
  >
    <div
      class="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
        style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
      />
      <div
        class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
        style="margin-bottom: 0px;"
      >
        <div
          aria-label="instruments.details.tabs_label"
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-controls="instrument-details-panel"
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
            id="instrument-details-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            instruments.details.label
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-11yukd5-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      aria-labelledby="instrument-details-tab"
      hidden=""
      id="instrument-details-panel"
      role="tabpanel"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-jddpo1-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            <b>
              Test Instrument
            </b>
          </h6>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-k7lt0k-MuiDivider-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            instruments.details.description
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              An instrument purely for testing
            </b>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            instruments.details.type
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              testing
            </b>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            instruments.details.url
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="www.example.com"
              >
                www.example.com
              </a>
            </b>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Instrument details panel component should render users tab when present in the data 1`] = `
<DocumentFragment>
  <div
    id="details-panel"
    style="min-width: 0;"
  >
    <div
      class="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
        style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
      />
      <div
        class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
        style="margin-bottom: 0px;"
      >
        <div
          aria-label="instruments.details.tabs_label"
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-controls="instrument-details-panel"
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
            id="instrument-details-tab"
            role="tab"
            tabindex="0"
            type="button"
          >
            instruments.details.label
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="instrument-users-panel"
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
            id="instrument-users-tab"
            role="tab"
            tabindex="-1"
            type="button"
          >
            instruments.details.instrument_scientists.label
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-11yukd5-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      aria-labelledby="instrument-details-tab"
      hidden=""
      id="instrument-details-panel"
      role="tabpanel"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-jddpo1-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            <b>
              Test Instrument
            </b>
          </h6>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-k7lt0k-MuiDivider-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            instruments.details.description
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              An instrument purely for testing
            </b>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            instruments.details.type
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              testing
            </b>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
          >
            instruments.details.url
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="www.example.com"
              >
                www.example.com
              </a>
            </b>
          </p>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="instrument-users-tab"
      hidden=""
      id="instrument-users-panel"
      role="tabpanel"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-jddpo1-MuiGrid-root"
      >
        <span
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          instruments.details.instrument_scientists.name
        </span>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              Louise Davies
            </b>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <b>
              Louise
            </b>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
